### DISCUSSION TITLE
Build at Runtime

### DISCUSSION THREAD

**Discussion by mariuslian** [vikejs/vike#1399] [Discussion]
Hi! In our Vue SPA users can build and publish sites and this is currently done throught rendering routes through a Page.vue component by passing in a data model that then renders dynamic components. It works but every route is essentially a SPA that is loaded each time. This isn't efficient or SEO friendly. 

What we want to do is this:
- After creating a site (which is an array of objects) a user clicks on "publish".
- We then process the pages which has paths and everthing that is needed.
- Each page is a route with an unique id that renders all the dynamic component like:
example.com/my-page -> Page.vue with pageId prop -> dynamic components creates the HTML output.
- In the end we have a set of routes with it's own output generated form Page.vue.
- They are then pushed to AWS.
- This could be Prerendered/static or SSR.

My understanding here is that for this to work:
- We won't add Vike plugin to our app vite.config.js since it is not our app we want to SSR but sites created on our app.
- Instead we need to create everything 100% programmatically and do the build process on the server at runtime.

My question is if this is a use case for Vike? 

Thanks,
Marius

**Comment by brillout** [comment]
AFAICT I don't foresee any blockers. So I'd suggest trying while letting us us know if you hit any blockers. Also see https://vike.dev/faq#can-i-use-vike-to-achieve-what-i-want.

**Reply by mariuslian** [reply]
@brillout my question was premature so that's why I closed it. I didn't get how flexible Vike was in the beginning but this is a very powerful tool and I'm now confident it will solve the mentioned use case.

**Reply by brillout** [reply]
👍 Btw. would your company be up for [sponsoring](https://github.com/vikejs/vike/issues/1350)?

**Reply by mariuslian** [reply]
Sponsorship added as an individual 👍

**Reply by lourot** [reply]
Thank you so much @mariuslian!

**Reply by brillout** [reply]
Thank you @mariuslian 💚
